qcodes/instrument_drivers/AlazarTech/utils.py

```diff
@@ -20,18 +25,22 @@ class TraceParameter(Parameter):
     an Alazar card is relatively slow, hence it makes sense to first set the
     values of the parameters, and then "synchronize them to the card".
     """
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any):
         self._synced_to_card = False
         super().__init__(*args, **kwargs)
 
-    def _set_updated(self):
+    def _set_updated(self) -> None:
         self._synced_to_card = True
 
     @property
     def synced_to_card(self) -> bool:
         """True if the parameter value has been synced to the instrument"""
         return self._synced_to_card
 
-    def set_raw(self, value):
-        self._instrument._parameters_synced = False
+    def set_raw(self, value: ParamRawDataType) -> None:
+        if TYPE_CHECKING:
+            instrument = cast("AlazarTech_ATS", self._instrument)
```

Comment:
if a string is used in the cast, is it really needed to be inside the `if TYPE_CHECKING` guard?